tools/merger/OpenApiMerger.ts

```diff
@@ -119,4 +122,23 @@ export default class OpenApiMerger {
     const gen = new SupersededOpsGenerator(this.root_folder)
     gen.generate(this.spec)
   }
+
+  filter_methods (paths: any): void {
+    for (const x_include_value of this.options.x_include) {
+      this.#filter_methods_with_key(paths, x_include_value)
+    }
+  }
+
+  #filter_methods_with_key (paths: any, key: string): void {
+    Object.entries(this.spec.paths as Document).forEach(([path, path_item]) => {
+      Object.entries(path_item as Document).forEach(([method, method_spec]) => {
+        if (method_spec[key] === undefined) {
+          delete (this.spec.paths[path][method])
+          if (Object.keys(this.spec.paths[path] as Document).length === 0) {
+            delete (this.spec.paths[path])
+          }
+        }
+      })
```

Comment:
@nhtruong if you want to help me rewrite this without `delete` I would gladly re-enable `no-dynamic-delete` - my typescript-fu did not cut it with `.filter` and friends.

Reply:
It's a good lint to have to prevent accidental deletions. But I think it's more prevalent to cases where the feature is heavily influenced  by the user input, as it's rather unpredictable. If you know that this has a very low risk of accidental deletions, we can disable the lint for the specific line like [this](https://github.com/opensearch-project/opensearch-api-specification/blob/52463cedb11012b33478c7ae01efb90adb18dd9c/tools/helpers.ts#L40) and leave the lint rule on.

Reply:
@Xtansia helped me with the code, newer revisions put this back